source/php-frameworks/symfony.txt

```diff
@@ -0,0 +1,372 @@
+   Symfony, see `Connecting to MongoDB
+   <https://www.mongodb.com/docs/php-library/current/tutorial/connecting/>`__
+   in the MongoDB PHP Library documentation.
+
+MongoDB Atlas is a fully managed cloud database service that hosts your
+MongoDB deployments. You can create your own free (no credit card
+required) MongoDB Atlas deployment by following the steps in this guide.
+
+This guide uses the MongoDB Doctrine ODM to allow you to map PHP objects to
+MongoDB documents and query MongoDB by using a builder API. 
+
+Follow the steps in this guide to create a sample Symfony web application
+that connects to a MongoDB deployment and performs a query on the
+database.
+
+Prerequisites
+~~~~~~~~~~~~~
+
+To create the Quick Start application, you need the following software
+installed in your development environment:
+
+- `PHP <https://www.php.net/downloads>`__
+- `Composer <https://getcomposer.org/download/>`__
+- `Symfony CLI <https://symfony.com/download>`__
+- A terminal app and shell. For MacOS users, use Terminal or a similar app.
+  For Windows users, use PowerShell.
+
+.. _php-symfony-atlas-cluster:
+
+Create a MongoDB Atlas Cluster
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You must create a MongoDB cluster where you can store and manage your
+data. Complete the :atlas:`Get Started with Atlas </getting-started>`
+guide to set up a new Atlas account and create a free tier MongoDB
+cluster. This tutorial also demonstrates how to load sample datasets
+into your cluster, including the data that is used in the Quick Start tutorial.
+
+You provide instructions to the driver on where and how to connect to your
+MongoDB cluster by providing it with a connection string. To retrieve
+your connection string, follow the instructions in the :atlas:`Connect
+to Your Cluster </tutorial/connect-to-your-cluster/>` tutorial in the
+Atlas documentation.
+
+Save your connection string in a secure location.
+
+Install MongoDB Extension
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To learn more about installing the MongoDB extension, see `Installing the Extension
+<https://www.mongodb.com/docs/php-library/current/tutorial/install-php-library/>`__
+in the PHP Library Manual.
+
+Initialize a Symfony Project
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Run the following command to create a skeleton Symfony project called
+``restaurants``:
+
+.. code-block:: bash
+
+   composer create-project symfony/skeleton restaurants
+
+Install PHP Driver and Doctrine ODM
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Enter your project directory, then add the MongoDB PHP driver and the
+MongoDB Doctrine ODM bundle to your application. The bundle integrates
+the ODM library into Symfony so that you can read from and write objects
+to MongoDB. To learn more, see the :ref:`php-symfony-resources` section
+of this guide.
+
+Run the following commands to install the driver and the ODM:
+
+.. code-block:: bash
+
+   composer require mongodb/mongodb
+   composer require doctrine/mongodb-odm-bundle
```

Comment:
@alcaeus: If we're relying on the bundle to pull in the [doctrine/mongodb-odm](https://packagist.org/packages/doctrine/mongodb-odm) package, does it make sense to explicit require `mongodb/mongodb`? That's also a dependency and would be pulled in automatically.

Reply:
That's correct, there's no advantage to manually requiring the package. There are some tools that forbid using transitive dependencies (that is, using classes from mongodb/mongodb when it's only required through a different dependency), but strictly speaking it makes no difference. I'd suggest removing the explicit require statement to make the installation process look simpler.